vertical-pod-autoscaler/e2e/actuation.go

```diff
@@ -39,54 +39,80 @@ var _ = ActuationSuiteE2eDescribe("Actuation", func() {
 	ginkgo.It("stops when pods get pending", func() {
 
 		ginkgo.By("Setting up a hamster deployment")
-		c := f.ClientSet
-		ns := f.Namespace.Name
+		d := SetupHamsterDeployment(f, "100m", "100Mi", defaultHamsterReplicas)
 
-		cpuQuantity := ParseQuantityOrDie("100m")
-		memoryQuantity := ParseQuantityOrDie("100Mi")
+		ginkgo.By("Setting up a VPA CRD with ridiculous request")
+		SetupVPA(f, "9999", vpa_types.UpdateModeAuto)
 
-		d := NewHamsterDeploymentWithResources(f, cpuQuantity, memoryQuantity)
-		d, err := c.AppsV1().Deployments(ns).Create(d)
-		gomega.Expect(err).NotTo(gomega.HaveOccurred())
-		err = framework.WaitForDeploymentComplete(c, d)
+		ginkgo.By("Waiting for pods to be restarted and stuck pending")
+		err := assertPodsPendingForDuration(f.ClientSet, d, 1, 2*time.Minute)
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
-		ginkgo.By("Setting up a VPA CRD with ridiculous request")
-		config, err := framework.LoadConfig()
+	})
+
+	ginkgo.It("never applies recommendations when update mode is Off", func() {
+		ginkgo.By("Setting up a hamster deployment")
+		d := SetupHamsterDeployment(f, "100m", "100Mi", defaultHamsterReplicas)
+		cpuRequest := getCPURequest(d.Spec.Template.Spec)
+		podList, err := GetHamsterPods(f)
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		podSet := MakePodSet(podList)
 
-		vpaCRD := NewVPA(f, "hamster-vpa", &metav1.LabelSelector{
-			MatchLabels: map[string]string{
-				"app": "hamster",
-			},
-		})
+		ginkgo.By("Setting up a VPA CRD in mode Off")
+		SetupVPA(f, "200m", vpa_types.UpdateModeOff)
 
-		resourceList := apiv1.ResourceList{
-			apiv1.ResourceCPU: ParseQuantityOrDie("9999"), // Request 9999 CPUs to make POD pending
```

Comment:
I already miss this comment in the new code (wondered for a while what 9999 stands for).

Reply:
Ah, I didn't intend to remove it. Done